homework/tai_nguyen/lesson10/JS_questions.md

```diff
@@ -0,0 +1,235 @@
+    { name: 'Vaalserberg', height: 323, place: 'Netherlands' },
+    { name: 'Denali', height: 6168, place: 'United States' },
+    { name: 'Popocatepetl', height: 5465, place: 'Mexico' },
+    { name: 'Mont Blanc', height: 4808, place: 'Italy/France' },
+  ];
+
+  const headerNames = ['name', 'height', 'place'];
+
+  function buildHeader(headerNames) {
+    const tr = document.createElement('tr');
+    headerNames.forEach((name) => {
+      const th = document.createElement('th');
+      th.innerText = name;
+      tr.appendChild(th);
+    });
+
+    return tr;
+  }
+
+  function buildRow(row, headerNames) {
+    const tr = document.createElement('tr');
+    headerNames.forEach((name) => {
+      const td = document.createElement('td');
+      if (typeof row[name] === 'number') {
+        td.style.textAlign = 'right';
+      }
+      td.innerText = row[name];
+      tr.appendChild(td);
+    });
+
+    return tr;
+  }
+
+  function buildTable(headerNames, rows) {
+    const table = document.createElement('table');
+    const header = buildHeader(headerNames);
+    table.appendChild(header);
+
+    rows.forEach((row) => {
+      const rowElement = buildRow(row, headerNames);
+      table.appendChild(rowElement);
+    });
+
+    return table;
+  }
+
+  const mountainDiv = document.getElementById('mountains');
+  if (mountainDiv) {
+    const table = buildTable(headerNames, MOUNTAINS);
+    mountainDiv.appendChild(table);
+  }
+</script>
+```
+
+**Question 2:** Implement your own version of `getElementByTagName`
+
+```typescript
+function byTagName<T extends keyof HTMLElementTagNameMap>(
+  node: Node,
+  tagName: T
+) {
+  const tagNameUppercase = tagName.toUpperCase();
+  const res: HTMLElementTagNameMap[T][] = [];
+
+  function visit(node: Node) {
+    if (node.nodeName === tagNameUppercase) {
+      res.push(node as HTMLElementTagNameMap[T]);
+    }
+
+    if (node.hasChildNodes()) {
+      node.childNodes.forEach((child) => {
+        visit(child);
+      });
+    }
+  }
+
+  visit(node);
+
+  return res;
+}
+```
+
+**Question 3:**
+
+**Question 4:**
+
+```css
+input[type='range'] {
+  accent-color: #f44336;
+}
+```
+
+**Question 5:** What's the output?
+
+```javascript
+const myPromise = Promise.resolve(Promise.resolve('Promise'));
+
+function funcOne() {
+  setTimeout(() => console.log('Timeout 1!'), 0);
+  myPromise.then((res) => res).then((res) => console.log(`${res} 1!`));
+  console.log('Last line 1!');
+}
+
+async function funcTwo() {
+  const res = await myPromise;
+  console.log(`${res} 2!`);
+  setTimeout(() => console.log('Timeout 2!'), 0);
+  console.log('Last line 2!');
+}
+
+funcOne();
+funcTwo();
+```
+
+- A: Promise 1! Last line 1! Promise 2! Last line 2! Timeout 1! Timeout 2!
+- B: Last line 1! Timeout 1! Promise 1! Last line 2! Promise2! Timeout 2!
+- C: Last line 1! Promise 2! Last line 2! Promise 1! Timeout 1! Timeout 2!
+- D: Timeout 1! Promise 1! Last line 1! Promise 2! Timeout 2! Last line 2!
+
+> Answer C
```

Comment:
Explain this question, pls @trongtai37

Reply:
Sorry @thaycacac, my last week was busy and I have not has time to explain all the questions yet.
This PR is temporarily draft and I will inform you when finishing these JS questions ASAP.
Thank you for your early review.

Reply:
thanks @trongtai37

Reply:
Hi @thaycacac, I have finished the all JS questions.
Please review it for me.
:D